napari/components/_dims/model.py

```diff
@@ -1,143 +1,255 @@
+        super().__init__()
 
-        self.viewer = viewer
-        self.events = EmitterGroup(source=self,
-                                   auto_connect=True,
-                                   update_slider=Event)
-        # TODO: allow arbitrary display axis setting
-        # self.y_axis = 0  # typically the y-axis
-        # self.x_axis = 1  # typically the x-axis
+        # Events:
+        self.events = EmitterGroup(source=self, auto_connect=True, axis=None,
+                                   ndim=None)
 
-        # TODO: wrap indices in custom data structure
-        self.indices = [slice(None), slice(None)]
+        self.range = []
+        self.point = []
+        self.interval = []
+        self.mode = []
+        self.display = []
 
-        self._max_dims = 0
-        self._max_shape = tuple()
+        self.ndim = init_ndim
 
-        # update flags
-        self._child_layer_changed = False
-        self._need_redraw = False
-        self._need_slider_update = False
+    def __str__(self):
+        return "~~".join(map(str, [self.range, self.point, self.interval,
+                                   self.mode, self.display]))
 
-        self._recalc_max_dims = False
-        self._recalc_max_shape = False
+    @property
+    def ndim(self):
+        """Returns the number of dimensions
 
-        self._qt = QtDims(self)
+        Returns
+        -------
+        ndim : int
+            Number of dimensions
+        """
+        return len(self.point)
+
+    @ndim.setter
+    def ndim(self, ndim):
+        if ndim > self.ndim:
+            for i in range(self.ndim, ndim):
+                self.range.insert(0, (0.0, 1.0, 0.01))
+                self.point.insert(0, 0.0)
+                self.interval.insert(0, (0.3, 0.7))
+                self.mode.insert(0, DimsMode.POINT)
+                self.display.insert(0, False)
+
+            # Notify listeners that the number of dimensions have changed
+            self.events.ndim()
+
+            # Notify listeners of which dimensions have been affected
+            for axis_changed in range(ndim - self.ndim):
+                self.events.axis(axis=axis_changed)
+
+        elif ndim < self.ndim:
+            self.range = self.range[-ndim:]
+            self.point = self.point[-ndim:]
+            self.interval = self.interval[-ndim:]
+            self.mode = self.mode[-ndim:]
+            self.display = self.display[-ndim:]
+
+            # Notify listeners that the number of dimensions have changed
+            self.events.ndim()
 
     @property
-    def max_dims(self):
-        """int: Maximum tunable dimensions for contained images.
+    def displayed(self):
+        """Returns the displayed dimensions
+
+        Returns
+        -------
+        displayed : np.ndarray
+            Displayed dimensions
         """
-        return self._max_dims
+        displayed = [i for i, d in enumerate(self.display) if d is True]
+        return np.array(displayed)
```

Comment:
does this really need to be an array? if so use `np.fromiter`

Reply:
no it doesn't - making it a list